### ISSUE TITLE
Polygons being culled incorrectly in 2D

### ISSUE THREAD

**Issue by mramato** [CesiumGS/cesium#1552]
I have a CZML file (which I can share) that loads a bunch of Polygons at an altitude above the earth.  It works great in 3D mode but the polygons disappear when in 2D.  They do show up in Columbus View when zoomed out, but get culled when you try and zoom anywhere near them.  I've scene similar things with other geometry types, so I'm not sure if this is a polygon specific issue or a problem with Geometry on the whole.

**Comment by hpinkos** [comment]
Code example:
``` javascript
var viewer = new Cesium.Viewer('cesiumContainer');

viewer.entities.add({
    polygon : {
        hierarchy : Cesium.Cartesian3.fromDegreesArray([-115.0, 37.0,
                                                        -115.0, 32.0,
                                                        -107.0, 33.0,
                                                        -102.0, 31.0,
                                                        -102.0, 35.0]),
        material : Cesium.Color.RED,
        height: 3000000
    }
});
```

**Comment by mzschwartz5** [comment]
Investigating this issue. Some immediate observations:
- As alluded to above, this is no longer a problem for CV, only 2D.
- If you remove the height field on the polygon, it does show up.
- Same issue is present for RectangleGeometry, with either `height` or even `extrudedHeight`

edit - *Interestingly*, the issue does not occur for any height value. E.g. In @hpinkos's example above, if you reduce the height from 3000000 to 1000000, the polygon *does* appear - until you zoom in a bit, then it gets culled again. 

If you turn on show frustums in the cesium inspector (see sandcastle in next comment), the culling seems to coincide with specific frustums.

**Comment by mzschwartz5** [comment]
[Interesting sandcastle](https://sandcastle.cesium.com/#c=rVNNb9swDP0rgi91gEy2ExRFUjdom+QwoF2Bpdtl3kGRWVuYLBmSnNYb+t8ryx81sg3b0PFgi4/kAx8JUim0QQcGj6DQBRLwiNagWVXgzw7zT6hz11IYwgSokyn6kQiENAUBtzKFZV+w6xG8W28/bGebRDxPzhORiJYegzDMMNCYpKnvSBAqJa8zKdASdUBjOQNFFM1rNJCviTL2RcQcPyhZbCBTAPpKKVL7X95F0SkOp2h+Zr+vNP9qA83sjTThmaOZv5Vm5mii/0NjpX2djIgKYkAxwkczllwq/HG7GWXlwLLcLFF0GjbWBp7/vNlur8Nah5X+3Ub7Bmz3oes+HA2hB2eLMbhY/A5riltoGEAv/kj79c2nbZcxCA8H4a+yqbsadwL2aLohONeG3R9nXO4Bp1Ca/B60ucrs9WhzD1YfE7bIqKpJ7gf4ZECkftdMC7bOe6FLoEaqW/bExOTcm3qxNjWHVa/ykhWlVAZVivsYBwaKklt5OthX9BsYTLVuum5S42BcGqfsgFh6kXhHN554iHKitY08VJzv2HdIvFUc2PyfSrkkKRPZ3QEUJ3WTlkermxbEGMeBdX9daaTke6KOmF8A)

Two polygons at different heights. The higher up one gets culled as you zoom in, but the lower one doesn't. AND - if you comment out the higher polygon, then the lower one DOES get culled... strange. Using the inspector, it seems that the bounding sphere of either poly is the same and is calculated from the combination of the polys. Might be relevant to why having both in scene changes the culling behavior.

**Comment by mzschwartz5** [comment]
A couple more interesting observations:

Even when the example polygon appears to get culled, it's still passing this [isVisible](https://github.com/CesiumGS/cesium/blob/a9480d31fdd85217c385b3bfc1659ae83c8d975c/packages/engine/Source/Scene/View.js#L345-L347) check! (edit- not only this, but the `DrawCommand` is clearly present in `Scene.executeCommands`.  This leads me to think the polygon is getting culled on the GPU side. Maybe because of the shenanigans going on [here](https://github.com/CesiumGS/cesium/blob/a9480d31fdd85217c385b3bfc1659ae83c8d975c/packages/engine/Source/Scene/Scene.js#L2641-L2647))

The problem still appears to be the bounding volume though, because if I modify the code to *always* use the camera's frustum to bound the object (i.e. take [this chunk](https://github.com/CesiumGS/cesium/blob/a9480d31fdd85217c385b3bfc1659ae83c8d975c/packages/engine/Source/Scene/View.js#L383-L388) out of the `else` block so it always runs), it renders properly.

**Comment by mzschwartz5** [comment]
Got it - it was the bounding sphere after all!
[Here's the problematic line](https://github.com/CesiumGS/cesium/blob/a9480d31fdd85217c385b3bfc1659ae83c8d975c/packages/engine/Source/Scene/Primitive.js#L1969). In 2D, `x` is the direction into/out of the screen. This line is setting the bounding sphere's center.x to 0. Then, when we find the [near and far planes of this bounding sphere, from the camera's vantage point](https://github.com/CesiumGS/cesium/blob/a9480d31fdd85217c385b3bfc1659ae83c8d975c/packages/engine/Source/Scene/View.js#L349-L353), they're relative to this x = 0 centered bounding volume. 

Long story short, this ultimately means that we end up binning the polygon into the wrong frustum partitions. If I simply remove the line that sets x = 0, the polygon doesn't get culled in any zoom level, and a quick check shows that the frustum splits are reasonable (i.e. it really works, it doesn't just "solve" the issue by forcing some weird frustum behavior).
